plugins/commands.js

```diff
@@ -34,22 +45,37 @@ exports.plugin = {
             handler: async (request, h) => {
                 const { namespace, name, version } = request.params;
                 const id = `${namespace}-${name}-${version}`;
-
+                let response;
                 let value;
 
-                try {
-                    value = await cache.get(id);
-                } catch (err) {
-                    throw err;
-                }
-
-                if (!value) {
-                    throw boom.notFound();
-                }
+                if (usingS3) {
+                    try {
+                        value = await awsClient.getDownloadStream({ cacheKey: id });
+                        response = h.response(value);
+                        response.headers['content-type'] = 'application/octet-stream';
+                    } catch (err) {
+                        request.log([id, 'error'], `Failed to stream the s3: ${err}`);
+                        throw err;
+                    }
+                } else {
+                    try {
+                        value = await cache.get(id);
+                    } catch (err) {
+                        throw err;
+                    }
 
-                const response = h.response(Buffer.from(value.c.data));
+                    if (!value) {
+                        throw boom.notFound();
+                    }
 
-                response.headers = value.h;
+                    if (value.c) {
+                        response = h.response(Buffer.from(value.c.data));
+                        response.headers = value.h;
+                    } else {
+                        response = h.response(Buffer.from(value));
```

Comment:
where is this change coming from ?

Reply:
This condition is not required, removed it.